### DISCUSSION TITLE
race_control_messages

### DISCUSSION THREAD

**Discussion by McHouse** [theOehrly/Fast-F1#392] [Q&A] [answered]
this works perfectly with versions 2.3.x
```
Session.load(laps=True, telemetry=True, weather=True, messages=True, livedata=None)
Control_messages=ff1.api.race_control_messages(Session.api_path)
```

but with versions 3.x.x it gives
AttributeError: module 'fastf1' has no attribute 'api'

I have seen in the documentation it appears:
Warning
fastf1.api will be considered private in future releases and potentially be removed or changed.

What is now the way to access those race messages?

Thanks in advance.

**Comment by theOehrly** [comment, marked as answer]
This likely doesn't work because your imports are incorrect. 

But in any case, the way to access race control messages is through the ``Session`` object.

https://docs.fastf1.dev/core.html#fastf1.core.Session.race_control_messages

**Reply by McHouse** [reply]
hmm...thinking...
Eureka!. You have opened my mind and now I see as a whole, the reorganization that you have carried out.
Great job.
Apologies for my inexperience with python.
From now on I hope I don't get so confused.
This is two coffees.

**Reply by theOehrly** [reply]
> hmm...thinking...
> Eureka!. You have opened my mind and now I see as a whole, the reorganization that you have carried out.
> Great job.
> Apologies for my inexperience with python.
> From now on I hope I don't get so confused.
> This is two coffees.

All good 😅 In fact, I'm pretty sure I know what has caused those troubles for you and it's something I hadn't thought of when changing this. Also, ``fastf1.api`` was never intended to be public really, but I didn't communicate that properly. I hope everything works now for you.
